### DISCUSSION TITLE
New Feature | Auto-Model Selection

### DISCUSSION THREAD

**Discussion by prasadautomationtesting** [open-webui/open-webui#7605] [Ideas]
I propose adding a feature that **automatically selects a model** based on the user's input, eliminating the need for repeated model selection. This would allow users to focus on their tasks without having to choose a model every time, saving time and increasing productivity.

**Example Use Cases:**

1. When drafting an email, the system would automatically select "**Llama 3.2:1**b" as the default model.
2. When working on a Python coding task, the system would automatically select "**Codellama**" as the default model.
3. When working on a complex task that requires **both natural language understanding and coding expertise**, the system could automatically select a combination of models, such as "Llama 3.2:1b" for text generation and "Codellama" for code review and optimization.

This feature would provide a more **streamlined and efficient user experience**, allowing users to quickly get started on their tasks without having to **manually select a model every time**.

**Comment by thfai2000** [comment]
More related situations:

1. When I want to upload an image/screenshot and give a prompt, the response is something about the current model (e.g. llama3.1) doesn't understand or able to read the image content. We have to switch to use LlaVa model that can understand images.

2. I had try to use "Tool" (which is to get external data) in prompt and asked my model to understand the content and to give summary. But the result is bad and seems the content cannot be detected. I spent hours and finally find out my model (deepseek-coder-v2:16b) doesn't support Tool. I think if the system can auto-detect and warn the user, the experience will be better. It may be perfect if a proper model is auto selected.

thank you.

**Comment by c-nielson** [comment]
Curious if there's been any more thought given to this. With some more powerful models being available now (gpt-oss, reasoning models), it would be nice to have a way for open-webui to default to a more lightweight model (llama, gemma, etc.) unless it thinks it needs some more reasoning effort put in, e.g. the user asks for data analysis for has a complex task.

**Comment by jbeltran73-2** [comment]
It's a fascinating discussion. It would be great to have Openwebui to implement it.
